python/lsst/validate/drp/matchreduce.py

```diff
@@ -295,73 +314,95 @@ def _loadAndMatchCatalogs(repo, dataIds, matchRadius,
     allMatches : `lsst.afw.table.GroupView`
         GroupView object with matches.
-    safeSnr : `float`, optional
-        Minimum median SNR for a match to be considered "safe".
+    goodSnr : float, optional
+        Minimum median SNR for a match to be considered "good"; default 3.
+    safeSnr : float, optional
+        Minimum median SNR for a match to be considered "safe"; default 50.
+    safeExtendedness: float, optional
+        Maximum (exclusive) extendedness for sources or minimum (inclusive) if extended==True.
+    extended: bool, optional
+        Whether to select extended sources, i.e. galaxies.
+    goodSnrMax : float, optional
+        Maximum median SNR for a match to be considered "good"; default np.Inf.
+    safeSnrMax : float, optional
+        Maximum median SNR for a match to be considered "safe"; default np.Inf.
     """
+    if nameFluxKey is None:
+        nameFluxKey = "slot_ModelFlux" if extended else "base_PsfFlux"
     # Filter down to matches with at least 2 sources and good flags
     flagKeys = [allMatches.schema.find("base_PixelFlags_flag_%s" % flag).key
                 for flag in ("saturated", "cr", "bad", "edge")]
     nMatchesRequired = 2
 
-    psfSnrKey = allMatches.schema.find("base_PsfFlux_snr").key
-    psfMagKey = allMatches.schema.find("base_PsfFlux_mag").key
-    psfMagErrKey = allMatches.schema.find("base_PsfFlux_magErr").key
+    snrKey = allMatches.schema.find(f"{nameFluxKey}_snr").key
+    magKey = allMatches.schema.find(f"{nameFluxKey}_mag").key
+    magErrKey = allMatches.schema.find(f"{nameFluxKey}_magErr").key
     extendedKey = allMatches.schema.find("base_ClassificationExtendedness_value").key
 
-    def goodFilter(cat, goodSnr=3):
+    snrMin, snrMax = goodSnr, goodSnrMax
+
+    def extendedFilter(cat):
         if len(cat) < nMatchesRequired:
             return False
         for flagKey in flagKeys:
             if cat.get(flagKey).any():
                 return False
-        if not np.isfinite(cat.get(psfMagKey)).all():
+        if not np.isfinite(cat.get(magKey)).all():
             return False
-        psfSnr = np.median(cat.get(psfSnrKey))
-        # Note that this also implicitly checks for psfSnr being non-nan.
-        return psfSnr >= goodSnr
+        extendedness = cat.get(extendedKey)
+        return np.min(extendedness) >= safeExtendedness if extended else \
+            np.max(extendedness) < safeExtendedness
 
-    goodMatches = allMatches.where(goodFilter)
-
-    # Filter further to a limited range in S/N and extendedness
-    # to select bright stars.
-    safeMaxExtended = 1.0
-
-    def safeFilter(cat):
-        psfSnr = np.median(cat.get(psfSnrKey))
-        extended = np.max(cat.get(extendedKey))
-        return psfSnr >= safeSnr and extended < safeMaxExtended
-
-    safeMatches = goodMatches.where(safeFilter)
+    def snrFilter(cat):
+        # Note that this also implicitly checks for psfSnr being non-nan.
+        snr = np.median(cat.get(snrKey))
```

Comment:
If you're worried about SNR potentially being NaN, then you should calculate the median with `np.nanmedian`.
`np.median` does what you expect with Inf, because Inf is sortable.  But if there is a NaN in the array `np.median` will return NaN.

Reply:
If
```
a = np.array([1])
b = np.array([0])
```
`a/b` is `np.array([Inf])`
`b/b` is `np.array([NaN])`

Reply:
This comment was already in the code but I accidentally shifted it up one line; it was referring to the comparison to snrMin/Max. So the existing behavior is to get a NaN median if any of the SNRs are NaN and hence return False. Do we want to tolerate some fraction of the SNRs being NaN?

Reply:
No, I think it's fine to not expect NaNs and to not tolerate any.

The executed code is fine.  The comment likely just needs to be moved.